index.js

```diff
@@ -525,9 +525,15 @@ Router.prototype.find = function find (method, path, derivedConstraints) {
         if (matchedParameter === null) return null
         paramEndIndex = pathIndex + matchedParameter[1].length
       } else {
+        const childrenStartCodes = node.numberOfChildren > 0 ? Object.keys(node.children).map(label => label.charCodeAt(0)) : []
         for (; paramEndIndex < pathLen; paramEndIndex++) {
           const charCode = path.charCodeAt(paramEndIndex)
-          if (charCode === 47 || charCode === 45 || charCode === 46) {
+          // Always break on /
+          if (charCode === 47) {
+            break
+          }
+          // Params start where next children begins
+          if (childrenStartCodes.includes(charCode)) {
```

Comment:
Includes is _really_ slow, could you use a for loop?

Reply:
Thanks for the review, I will comeback with improvements